packages/core-mobile/app/screens/earn/Confirmation/Confirmation.tsx

```diff
@@ -107,6 +132,9 @@ export const Confirmation = (): JSX.Element | null => {
     delegationFee: Number(validator?.delegationFee)
   })
 
+  const feeState = getFeeState()
+  const excessiveNetworkFee =
+    !!feeState?.price && !!gasPrice && gasPrice > feeState.price * 2n
```

Comment:
you should create a hook for this (useIsNetworkFeeExcessive or something) to reuse the logic. in there it can reuse the cached feeState